api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -387,18 +406,181 @@ public boolean supportsPropertyName(String gpName) {
 	}
 
 	@Override
-	public String importMpiPatient(String personId) {
-		if (registrationCoreProperties.isMpiEnabled()) {
-			MpiProvider mpiProvider = registrationCoreProperties.getMpiProvider();
-			Patient importedPatient = mpiProvider.fetchMpiPatient(personId);
-			Patient patient = patientService.savePatient(importedPatient);
-			return patient.getUuid();
+	public MpiPatient findMpiPatient(String identifier, String identifierTypeUuid) {
+		if (!registrationCoreProperties.isMpiEnabled()) {
+			throw new MpiException("Should not perform 'findMpiPatient' when MPI is disabled");
+		}
+		MpiProvider mpiProvider = registrationCoreProperties.getMpiProvider();
+		MpiPatient mpiPatient = null;
+		try {
+			mpiPatient = mpiProvider.fetchMpiPatient(identifier, identifierTypeUuid);
+		} catch (RuntimeException e) {
+			servePdqExceptionAndThrowAgain(e,
+					prepareParameters(identifier, identifierTypeUuid),
+					PdqErrorHandlingService.FIND_MPI_PATIENT_DESTINATION);
+		}
+		return mpiPatient;
+		//TODO throw error if you get more than one patient
+	}
+
+	@Override
+	public Patient importMpiPatient(String personId) {
+		return importMpiPatient(personId, mpiProperties.getMpiPersonIdentifierTypeUuid());
+	}
+
+	@Override
+	public Patient importMpiPatient(String patientIdentifier, String patientIdentifierTypeUuid) {
+		MpiPatient foundPatient = findMpiPatient(patientIdentifier, patientIdentifierTypeUuid);
```

Comment:
This is where you should be calling `mpiProperties.getMpiPersonIdentifierTypeUuid()`

Reply:
If we do this we will remove the flexibility to import an mpi patient not based on the MPI global domain identifier, (as mentioned above) do we want to do this?